semantic/poly_types.go

```diff
@@ -761,8 +761,17 @@ func (l ObjectKind) unifyKind(kinds map[Tvar]Kind, k Kind) (Kind, Substitution,
 	upper := l.upper.intersect(r.upper)
 	lower := l.lower.union(r.lower)
 
+	// Ensure that all of the values that are missing are allowed to be missing.
 	diff := lower.diff(upper)
-	if len(diff) > 0 {
+	for _, lbl := range diff {
+		if ptv, ok := properties[lbl].(Tvar); ok {
+			// If this tvar is nullable, then it is allowed
+			// to be missing.
+			kind := kinds[ptv]
+			if _, ok := kind.(NullableKind); ok {
```

Comment:
When will this not be the case? All member expressions induce a nullable kind constraint on the property type. Are we certain that kind will be removed from the `kinds` map at this point?

Reply:
Yes because `unifyKinds` has been modified to require all `Tvar` values it depends on to have already been resolved. So if a `NullableKind` constraint is added to the same `Tvar` that is also an `ObjectKind`, it will be resolved as an `ObjectKind`. This is the part that causes the added test to succeed.

Reply:
You can see the line getting hit in code coverage: https://codecov.io/gh/influxdata/flux/pull/1363/src/semantic/poly_types.go?before=semantic/poly_types.go#L775